src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java

```diff
@@ -407,6 +417,60 @@ private Link linkToFullImage(long imageId) {
         return fullImageLink;
     }
 
+    /**
+     * Generates a vaadin Button which when pressed will retrieve the metadata information of the given imageId  from the omero server and present it in a seperate subwindow
+     * @param imageId the id of the image for which the metadata should be retrieved
+     * @return a vaadin {@Button Button} component
+     */
+    private Button metadataButton(long imageId) {
+        Button metadataButton = new Button("Show Metadata");
+        metadataButton.setEnabled(false);
+        Collection<MetadataProperty> metadataProperties = new ArrayList<>();
+        try {
+            List metadataList = omeroClient.fetchMapAnnotationDataForImage(imageId);
+            for (int i = 0; i < metadataList.size(); i++) {
+                MapAnnotationData currentMapAnnotation = (MapAnnotationData) metadataList.get(i);
+                List<NamedValue> list = (List<NamedValue>) currentMapAnnotation
+                    .getContent();
+                for (NamedValue namedValue : list) {
+                    String metaDataKey = namedValue.name;
+                    String metaDataValue = namedValue.value;
+                    metadataProperties.add(new MetadataProperty<String>("Key: " + metaDataKey, "Value " + metaDataValue, "example property no."));
+                    metadataButton.setEnabled(true);
+                }
+            }
+
+        } catch (Exception e) {
+            LOG.error("Could not retrieve metadata for image:" + imageId);
+            LOG.debug(e);
+
+        }
+        metadataButton.addClickListener(clickEvent -> {
```

Comment:
You again have an event-based action here. Wrap the click event body in a try-catch block.

Reply:
Done!